### PR TITLE
Homework10: Alexey Kozhan

### PR DESCRIPTION
**Pull request by AlexKozhan**
Homework is done in 4 python files

### CODE REVIEW

File: homework1.py [hunk 1]
```diff
@@ -1,33 +1,32 @@
...
-""              ==>  ""
-"""
 
+def validate_arguments(func):
+    """Decorator that checks that all
+    function arguments are positive numbers."""
+    def wrapper(*args):
+        for i in args:
+            if i <= 0:
+                raise ValueError('Введите положительные числа')
```
Comment (Stml89) [inline homework1.py:21]:
```suggestion
            if i < 0:
                raise ValueError('Числа должны быть положительными')
```

File: homework4.py
Comment (Stml89) [inline homework4.py:1]: Nice!

File: homework2.py [hunk 2]
```diff
@@ -1,53 +1,39 @@
...
+
+
+def validate_type(func):
+    """Decorator that checks the type of a function's result."""
+    def wrapper(*args):
+        inner_result = func(*args)
+        if not isinstance(inner_result, (int, float)):
+            print(f"Result of function is "
+                  f"not a number, instead it's {type(inner_result)}")
+            return None
```
Comment (Stml89) [inline homework2.py:18]: return False

File: homework2.py
Comment (Stml89) [inline homework2.py:1]: You are using the same filename, so git shows the difference. Please, create files with different filenames

**Review by Stml89 — APPROVED** [review]

File: homework1.py (hunk 1 above)
Comment (AlexKozhan) [reply homework1.py]: fixed

File: homework2.py (hunk 2 above)
Comment (AlexKozhan) [reply homework2.py]: fixed

File: homework2.py
Comment (AlexKozhan) [reply homework2.py:1]: fixed

File: homework4.py
Comment (AlexKozhan) [reply homework4.py:1]: ty

File: resultati.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+@validate_type
+def plus_numbers(a, b):
+    """Function for plussing two numbers."""
+    return a + b
+
+
+@validate_type
+def plus_strings(a, b):
+    """Function for concatenating two strings."""
+    return a + b
```
Comment (Stml89) [inline resultati.py:32]: 1 function should work the same for numbers and letters
```suggestion
@validate_type
def plus_two_value(a, b):
    """Function for concatenating two values"""
    return a + b
```

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: resultati.py (hunk 3 above)
Comment (AlexKozhan) [reply resultati.py]: fixed

**Review by Stml89 — APPROVED** [review]

**Pull request merged**